src/CUDA/ddot.jl

```diff
@@ -9,5 +9,6 @@
     @inbounds for i=1:n
         v += dx[i]*dy[i]
     end
+    CUDA.sync_threads()
```

Comment:
Wasn't there `@synchronize`? It seems that in the main branch `@synchronize` has been placed there.